includes/namespace.php

```diff
@@ -79,5 +79,6 @@ function deactivate( $network_wide ) {
 register_activation_hook( WEBSTORIES_PLUGIN_FILE, '\Google\Web_Stories\activate' );
 register_deactivation_hook( WEBSTORIES_PLUGIN_FILE, '\Google\Web_Stories\deactivate' );
 
+global $web_stories;
```

Comment:
Making this global is important here. 

To unhook a class, should be simple as. 

```
global $web_stories;
remove_action('init', [$web_stories->media, 'init']);
```

Not the cleanest, but surely possible to do now. We have to document this somewhere. Thoughts @swissspidy

Reply:
Maybe make it a singleton instead of using a global?

Example from Site Kit:

https://github.com/google/site-kit-wp/blob/develop/includes/loader.php#L60-L61

https://github.com/google/site-kit-wp/blob/ff9c5141800adbad9a1cf26ca09b291a57bd361a/includes/Plugin.php#L205-L233

Reply:
I don't love it being a global, but singleton are really bad OOP. 
As it is not a singleton we can do this in a test.
```
public function test_plugin(){
    $plugin = new Plugin();
    $result = $plugin->test_method( '123', '567');
    ....
}
```

Singleton make it hard to manage state and mock in classes.
There is a lot of reading on why singletons are a bad idea.
- https://www.davidtanzer.net/david's%20blog/2016/03/14/6-reasons-why-you-should-avoid-singletons.html
- https://sergeyzhuk.me/2017/07/11/singleton/
- http://blog.gordon-oheim.biz/2011-01-17-Why-Singletons-have-no-use-in-PHP/